### PR TITLE
[Feature] Build Completion Footer for Processing Tasks

### PR DESCRIPTION
**Pull request by aaggarwal10**
## Notion ticket link

[Add Footer for completing request](https://www.notion.so/uwblueprintexecs/ac7b1f636a214a0cbf45f6b76f899b54?v=5414ee5fc5974f149921e192d6eb29b8&p=52eb3c79cb4d4097bc08bf701afc965c)

## Implementation description
* Added Footer using `Chakra UI` with `Complete request` button and `View permit holder`
* Removed old complete request button
* Hooked up View permit holder with custom GraphQL query
## Checklist
- [X] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [X] I have run the appropriate linter(s)
- [X] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

**Review by chrischan325 — APPROVED** [review]
Everything looks good to me, assuming the backend processes of completing request is not included in this ticket

File: tools/admin/requests/view-request.ts [hunk 1]
```diff
@@ -52,3 +53,22 @@ export type GetApplicationResponse = {
...
+export const GET_APPLICATION_APPLICANT = gql`
+  query GetApplicationProcessing($id: Int!) {
+    application(id: $id) {
+      applicant {
+        id
+      }
+    }
+  }
+`;
+
```
Comment (OustanDing) [inline tools/admin/requests/view-request.ts:67]: This query is not necessary. One of the main benefits of GraphQL is that you can query for whatever you need in just one query. Thus, you can select `applicant > id` from the application in the `GET_APPLICATION_QUERY` above.

File: pages/admin/request/[id].tsx [hunk 2]
```diff
@@ -26,11 +30,22 @@ type Props = {
  */
 const Request: NextPage<Props> = ({ id: idString }: Props) => {
   const id = parseInt(idString);
-
+  const [permitHolderID, setPermitHolderID] = useState<number | undefined>(undefined);
```
Comment (OustanDing) [inline pages/admin/request/[id].tsx:33]: This state variable isn't needed. After querying for the applicant ID in the existing GQL query in this file, just destructure it from `data`

**Review by OustanDing — CHANGES_REQUESTED** [review]

**Comment by OustanDing** [comment]
Also, good idea to use `Portal` to build the footer! We should use that across our app in general

File: components/admin/requests/Header.tsx [hunk 3]
```diff
@@ -59,9 +63,64 @@ export default function RequestHeader({
         );
       case 'IN_PROGRESS':
         return (
-          <CompleteRequestModal applicationId={applicationId}>
-            <Button disabled={!allStepsCompleted}>Mark as complete</Button>
-          </CompleteRequestModal>
+          <Portal>
```
Comment (OustanDing) [inline components/admin/requests/Header.tsx:66]: Can we extract this to a component named `ProcessingFooter` and put it in `components/admin/requests`?

File: components/admin/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { Portal, Box } from '@chakra-ui/react'; // Chakra UI
+import { ReactNode } from 'react'; // React
+
+type ProcessingTasksFooterProps = {
+  readonly children: ReactNode;
+};
+
+export default function ProcessingTasksFooter({ children }: ProcessingTasksFooterProps) {
```
Comment (OustanDing) [inline components/admin/Footer.tsx:8]:
```suggestion
export default function Footer({ children }: ProcessingTasksFooterProps) {
```
We will use this component as a generic footer wrapper

File: components/admin/requests/Header.tsx [hunk 5]
```diff
@@ -40,7 +38,7 @@ export default function RequestHeader({
   createdAt,
   allStepsCompleted,
   applicationType,
-  permitHolderId,
+  applicantId: permitHolderId,
```
Comment (OustanDing) [inline components/admin/requests/Header.tsx:41]:
```suggestion
  applicantId,
```

File: components/admin/requests/Header.tsx [hunk 6]
```diff
@@ -63,64 +61,11 @@ export default function RequestHeader({
...
-                      </a>
-                    </Link>
-                  </Stack>
-                </Box>
-              </Stack>
-            </Box>
-          </Portal>
+          <ProcessingTasksFooter
+            applicationId={applicationId}
+            permitHolderId={permitHolderId}
```
Comment (OustanDing) [inline components/admin/requests/Header.tsx:66]:
```suggestion
            permitHolderId={applicantId}
```

File: components/admin/requests/processing/ProcessingFooter.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
+import { Button, Text, Box, Stack } from '@chakra-ui/react'; // Chakra UI
+import Link from 'next/link'; // Link
+import CompleteRequestModalButton from '@components/admin/requests/processing/CompleteModalButton'; // Mark as complete button + modal
+import Footer from '@components/admin/Footer';
+type ProcessingTasksFooterProps = {
+  readonly applicationId: number;
+  readonly permitHolderId?: number;
```
Comment (OustanDing) [inline components/admin/requests/processing/ProcessingFooter.tsx:7]:
```suggestion
  readonly applicantId?: number;
```

File: components/admin/requests/processing/ProcessingFooter.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  readonly allStepsCompleted: boolean;
+};
+
+export default function ProcessingTasksFooter({
+  applicationId,
+  permitHolderId,
```
Comment (OustanDing) [inline components/admin/requests/processing/ProcessingFooter.tsx:13]:
```suggestion
  applicantId,
```

File: components/admin/requests/processing/ProcessingFooter.tsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        <Box>
+          <Stack direction="row" justifyContent="space-between">
+            <Link href="#">
+              <a>
+                <CompleteRequestModalButton
+                  applicationId={applicationId}
+                  isDisabled={!allStepsCompleted}
+                />
+              </a>
+            </Link>
```
Comment (OustanDing) [inline components/admin/requests/processing/ProcessingFooter.tsx:49]:
```suggestion
                <CompleteRequestModalButton
                  applicationId={applicationId}
                  isDisabled={!allStepsCompleted}
                />
```
This should not be wrapped in `Link` as it does not navigate to a new page route

File: pages/admin/request/[id].tsx [hunk 10]
```diff
@@ -30,27 +26,17 @@ type Props = {
...
-      notifyOnNetworkStatusChange: true,
-    }
-  );
   // Get Permit Holder ID from Application
 
   if (!data?.application) {
     return null;
   }
 
+  const permitHolderID = data.application.applicant?.id;
```
Comment (OustanDing) [inline pages/admin/request/[id].tsx:39]:
```suggestion
  const applicantId = data.application.applicant?.id;
```

**Review by OustanDing — APPROVED** [review]
LGTM!

Design approved by Vedant as well.

**Pull request merged**